### ISSUE TITLE
Print Sequence by Grouped Objects

### ISSUE THREAD

**Issue by MrTyton** [bambulab/BambuStudio#4716]
Currently, there are two ways to print items - either by layer, or by object.

However, take the case where we have multiple small objects, with groups of them that are the same color, such as here

![image](https://github.com/user-attachments/assets/3224845c-1e20-4842-a2e7-ffd8f0062de2)

There are 3 ways to tackle this -

1) Print by layer on multiple build plates, which requires setup/removal time between each batch

2) Print by layer on one build plate, which can let you have a large number of these groups, but wastes filament and time through repeated filament changes.

3) Print by Object, which allows you to have minimal filament changes, but also minimizes the number of items that can be present on the build plate, since the items have to be arranged so that they never hit the print head (clearance issues).

I am proposing a 4th mode, that is not implemented, which is "Print by Layered Groups".

Basically, since the items can be grouped by the filament type, there's no reason that you shouldn't be able to print those layer by layer, like normal, and leave clearance for _another_ group, with a different filament type, that can then be printed layer by layer.

![image](https://github.com/user-attachments/assets/31e3bc0f-bc30-4375-83d6-658db67dd620)

In that picture above, the 2 groups have enough clearance that working on one group will not interfere with the other group, and the clearance of the items within each of the groups doesn't matter since it is done print by layer.

This would allow for more groups to be put onto the plate, which would printing time and user time, due to the reasons stated above.

**Comment by hadess** [comment]
https://github.com/bambulab/BambuStudio/issues/5501 has a simpler "print by color" which would be less complicated to implement UI-wise.
